src/main/java/org/cirdles/topsoil/dataset/DatasetResourceManager.java

```diff
@@ -0,0 +1,108 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.cirdles.topsoil.app.dataset.reader.DatasetReader;
+import org.cirdles.topsoil.app.dataset.reader.TSVDatasetReader;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class DatasetResourceManager {
+
+    private final List<DatasetResource> datasetResources;
+
+    public DatasetResourceManager(Path basePath) {
+
+        datasetResources = getInitialDatasetResources(basePath);
+
+        try {
+            Files.createDirectories(basePath);
+        } catch (IOException ex) {
+            Logger.getLogger(DatasetResourceManager.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+
+    public void addDatasetResources(DatasetResource datasetResource) {
+        datasetResources.add(datasetResource);
+    }
+
+    public void removeDatasetResources(DatasetResource datasetResource) {
+        datasetResources.remove(datasetResource);
+    }
+
+    public List<DatasetResource> getDatasetResources() {
+        return datasetResources;
+    }
+
+    private List<DatasetResource> getInitialDatasetResources(Path basePath) {
+        List<DatasetResource> resources = new ArrayList();
+
+        resources.addAll(getDatasetsForPath(basePath.resolve("open")));
+        resources.addAll(getDatasetsForPath(basePath.resolve("closed")));
+
+        return resources;
+    }
+
+     private List<DatasetResource> getDatasetsForPath(Path path) {
+        List<DatasetResource> resources = new ArrayList<>();
+        DatasetReader datasetReader = new TSVDatasetReader(true);
+
+        try {
+            // create the path if it does not yet exist
+            Files.createDirectories(path);
+
+            Files.walk(path, 1).forEach(datasetPath -> {
+                try {
+                    if (Files.isDirectory(datasetPath)) {
+                        return;
+                    }
+
+                    SimpleDataset dataset = (SimpleDataset) datasetReader.read(datasetPath);
+                    resources.add(DatasetResourceFactory.getDatasetResource(dataset, datasetPath, path.getParent()));
+
+                    dataset.setName(datasetPath.getFileName().toString());
+                } catch (IOException ex) {
+                    Logger.getLogger(DatasetResourceManager.class.getName()).log(Level.SEVERE, null, ex);
+                }
+            });
+        } catch (IOException ex) {
+            Logger.getLogger(DatasetResourceManager.class.getName()).log(Level.SEVERE, null, ex);
+        }
+
+        return resources;
+    }
+
+    public Optional<DatasetResource> getDatasetResource(Dataset dataset) {
+        Optional<DatasetResource> datasetResource = Optional.empty();
+        for (int i = 0 ; i < datasetResources.size(); i++) {
+            if (datasetResources.get(i).getDataset() == dataset) {
+                datasetResource = Optional.of(datasetResources.get(i));
+                break;
+            }
+        }
+        return datasetResource;
+    }
+}
```

Comment:
Creating directories can be the responsibility of the `DatasetResource`, but I think you're right that there could be a purpose for a class like this in loading up old `DatasetResource`s. The trick is to make sure that it's _only_ coordinating the loading of the resources (maybe have it call something a special constructor or method on `DatasetResourceFactory` rather than parsing files directly). It shouldn't have any sort of add method and shouldn't ever take or give `Dataset`s. Think of it more as a `DatasetResourceLoader` than as a manager. `DatasetResource`s _manage themselves_ after they're created. That's the beauty. They handle all the CRUD operations.